core/api/src/jsonrpc/web3_types.rs

```diff
@@ -97,15 +97,21 @@ impl From<SignedTransaction> for Web3Transaction {
             )
         };
 
+        let gas = match stx.transaction.unsigned {
+            UnsignedTransaction::Legacy(ref inner) => inner.gas_limit,
+            UnsignedTransaction::Eip1559(ref inner) => inner.gas_limit,
+            UnsignedTransaction::Eip2930(ref inner) => inner.gas_limit,
+        };
+
         Web3Transaction {
-            type_:                    Some(stx.type_().into()),
-            block_number:             None,
-            block_hash:               None,
-            raw:                      Hex::encode(stx.transaction.encode().unwrap()),
-            public_key:               stx.public,
-            gas:                      U256::zero(),
-            gas_price:                stx.transaction.unsigned.gas_price(),
-            max_fee_per_gas:          if is_eip1559 {
+            type_: Some(stx.type_().into()),
+            block_number: None,
+            block_hash: None,
+            raw: Hex::encode(stx.transaction.encode().unwrap()),
+            public_key: stx.public,
+            gas,
```

Comment:
For such a big struct that has so many fields, I prefer the following code:
```suggestion
            gas: gas,
```

Then the diff will be very small after formatting.